### PR TITLE
Added optional compile warnings + some fixes

### PR DESCRIPTION
**Pull request by a-w50**
- add possibility to specify compiler warnings
- in order to switch on compile warnings for source files, use cmake ... -DCOMPILER_WARNING_OPTIONS="-Wall;-Wextra;-Wpedantic" this will add these flags to all source files of this repository targets
- a few of the resulting warnings have been resolved, some are still there and need to be fixed

### CODE REVIEW

File: src/controller/server.cpp [hunk 1]
```diff
@@ -119,13 +119,32 @@ const lws_protocol_vhost_options Server::Impl::pvo = {nullptr, &pvo_opt, "everes
 
 const lws_protocol_vhost_options Server::Impl::pvo_mime = {nullptr, nullptr, ".mp4", "application/x-mp4"};
 
+// lws_http_mount Server::Impl::mount = {
```
Comment (hikinggrass) [inline src/controller/server.cpp]: why keep this around?

File: lib/serial.cpp [hunk 2]
```diff
@@ -117,6 +117,7 @@ uint32_t Serial::crc32(uint8_t* buf, int len) {
 }
 
 void Serial::handlePacket(uint8_t* buf, int len) {
+    // FIXME (aw): why is there no implementation here?
```
Comment (hikinggrass) [inline lib/serial.cpp:120]: we should make this virtual, probably does not make much sense to have a default implementation

File: src/controller/server.cpp (hunk 1 above)
Comment (a-w50) [reply src/controller/server.cpp]: sorry, forgot to delete that

File: lib/serial.cpp (hunk 2 above)
Comment (a-w50) [reply lib/serial.cpp:120]: yes, I guess this should be done in a separate refactor for the serial lib.  Currently Yeti and uMWC for example still don't use it and have their own `openSerial` implementation

**Review by hikinggrass — APPROVED** [review]

**Pull request merged**